l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -697,18 +735,29 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         require(l2BridgeAddress[ERA_CHAIN_ID] != address(0), "ShB b. n dep");
         require(_l1Asset != L1_WETH_TOKEN, "ShB: WETH deposit not supported 2");
 
-        // // Note that funds have been transferred to this contract in the legacy ERC20 bridge.
-        // if (!hyperbridgingEnabled[ERA_CHAIN_ID]) {
-        //     chainBalance[ERA_CHAIN_ID][_l1Asset] += _amount;
-        // }
+        bytes32 _assetInfo;
+        bytes memory bridgeMintCalldata;
 
-        bytes memory bridgeMintData = abi.encode(_amount);
-        bytes memory l2TxCalldata = _getDepositL2Calldata(
-            IL1NativeTokenVault(nativeTokenVault).getAssetInfoFromLegacy(_l1Asset),
-            bridgeMintData
-        );
+        {
+            bytes memory _assetData;
+
+            // Inner call to encode data to decrease local var numbers
+            try this.decodeLegacyData(abi.encode(_l1Asset, _amount, _l2Receiver), _prevMsgSender) returns (
```

Comment:
This is not needed, you are encoding and decoding the same data, why?

Reply:
updated